Core/include/Acts/Utilities/Intersection.hpp

```diff
@@ -175,4 +177,60 @@ struct SameSurfaceIntersection {
   }
 };
 
+namespace detail {
+
+/// This function checks if an intersection is valid for the specified
+/// path-limit and overstep-limit
+///
+/// @tparam intersection_t Type of the intersection object
+/// @tparam logger_t The logger type, which defaults to std::false_type to
+/// prevent the generation of logging code
+///
+/// @param intersection The intersection to check
+/// @param pLimit The path-limit
+/// @param oLimit The overstep-limit
+/// @param logger A optionally supplied logger which prints out a lot of infos
+/// at VERBOSE level
+template <typename intersection_t, typename logger_t = std::false_type>
+bool checkIntersection(const intersection_t& intersection, double pLimit,
+                       double oLimit, logger_t logger = logger_t{}) {
```

Comment:
I would probably prefer having the tolerance be an explicit parameter here. What do you think @benjaminhuth?

Reply:
Good idea!